packages/sources/cryptocompare-test/src/endpoint/crypto-ws.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  AdapterContext,
+  PriceEndpoint,
+  PriceEndpointParams,
+} from '@chainlink/external-adapter-framework/adapter'
+import {
+  WebSocketTransport,
+  WebSocketRawData,
+  WebSocket,
+} from '@chainlink/external-adapter-framework/transports/websocket'
+import { ProviderResult, makeLogger } from '@chainlink/external-adapter-framework/util'
+import { DEFAULT_WS_API_ENDPOINT } from '../config'
+import { cryptoInputParams } from '../crypto-utils'
+
+interface Message {
+  PRICE?: number // Cryptocompare does not provide the price in updates from all exchanges
+  TYPE: string
+  MARKET: string
+  FLAGS: number
+  FROMSYMBOL: string
+  TOSYMBOL: string
+  VOLUMEDAY: number
+  VOLUME24HOUR: number
+  VOLUMEDAYTO: number
+  VOLUME24HOURTO: number
+  MESSAGE?: string
+}
+
+export interface WSErrorType {
+  TYPE: string
+  MESSAGE: string
+  PARAMETER: string
+  INFO: string
+}
+
+const logger = makeLogger('CryptoCompareCryptoEndpoint')
+
+export const transport = new WebSocketTransport({
+  url: (context: AdapterContext) =>
+    `${DEFAULT_WS_API_ENDPOINT}?api_key=${context.adapterConfig.API_KEY}`,
```

Comment:
This does not allow the ws api endpoint to be modified with the env var from the config, is this intended?

Reply:
This is how it was done in the v3 repo, so I didn't intend anything further than that. Is there a more ideal way to do this?

Reply:
This has been updated now. Please confirm if resolved

Reply:
It is, ty